apps/src/applab/applab.js

```diff
@@ -1315,6 +1326,21 @@ Applab.execute = function() {
     }
   }
 
+  if (Applab.makerlabController) {
+    Applab.makerlabController.ensureBoardConnected(
+        function (e) {
+          console.log("Error connecting to board.", e);
+        },
+        function () {
+          Applab.makerlabController.installComponentsOnInterpreter(codegen, Applab.JSInterpreter);
+          Applab.beginVisualizationRun();
+        });
```

Comment:
Sounds good, that does feel better.

I'd slightly prefer still explicitly passing the codegen/interpreter since they get modified at that time, and Applab is otherwise handling their lifecycle. If e.g. something else later extends the `customMarshalObjectList` and finds this stomps on the setting, might make it easier to track down what's happening.